### ISSUE TITLE
Bug - Use of 'connect the features' multiple times

### ISSUE THREAD

**Issue by arch0345** [openstreetmap/iD#8655]
This issue can be recreated by adding a marked crosswalk that crosses a dual carriageway. An issue should pop up with iD suggesting to 'connect the features'. This function works the first time, but when clicking on it again, the marked crosswalk node is only connected to the crosswalk way and not with the roadway.

**Comment by AntMadeira** [comment]
I can confirm this.
I found out that when crossing a way over an area, iD suggests to connect them, but only the first node works. The next one creates a disconnected point (white balloon).

**Comment by jtracey** [comment]
Issue I suspect is directly related: Draw a new highway over an existing, unmodified highway, such that they trigger the "crossing highways" warning. Then, move one of the nodes on the existing highway, so that where they cross is in a different position, but they still cross. Hit "Connect features", and an intersection node will be added at the *old* intersection location, rather than the current one.

For additional weirdness: Do the above, but instead of moving an existing node, create a new node on the existing highway, and move that. "Connect features" will add nodes that aren't connected to either way (untagged nodes if both ways are vehicle roads, crossing nodes if one is a pedestrian way).

**Comment by bhousel** [comment]
I did some investigation into this today.. It's related to a change I made in 1f172d5623921a2d84880ce705cdee3193a70df6 . 
The crossing issue id is less strict now, to avoid the situation where any slight adjustment of the way would "fix" the old issue and create a new one.   The problem with using a less strict id is that now d3 holds onto the "old" issue fix in the inspector window (it thinks nothing has changed), so when you click that fix button it tries to fix it the "old" way.  (If you select a different feature and then go back to the feature with the issue, d3 actually does recreate the dom and use the "current" issue and "current" fix).

One way to fix this is to make the issues work similar to how we manage OSM entities: each entity has both an `id` which is semi stable (`n123`), and a `key` with an internal version (`n123v1`) that changes enough for d3 to realize when things need a refresh.
